Documentation Update

TimPatrickProgramming opened the pull request:
- Updated The documentation in the README file.
- Added the neccesary diagrams written with mermaid

## Things to check

- Does the Script have a seperate GUI and console mode?
-- Meaning in one mode the GUI is engaded and in the other it is not

TimPatrickProgramming committed:
Created Documentation Branche

- Created the Branch for Documentation
- Reworked Purpose of this project
- Added Key Features

TimPatrickProgramming committed:
Added Project Setup

- Added Project Setup which explains the activation of the venv and which script needs to be executed

TimPatrickProgramming committed:
Added Interpreter Selection to README

-  Added Interpreter Selection to the project setup

TimPatrickProgramming committed:
Added Static View Class Diagram

- Added Shape Detector App class to the class diagram

TimPatrickProgramming committed:
Added 3 new classes

- Added 3 new classes to static view class diagram
- Image processing
- Image Loader
- Logger

TimPatrickProgramming committed:
Static view Class diagram finished

- Finished the Class diagram
- Added the Shape calss and its subcalsses
-- Triangle
-- Rectangle
-- Square
-- Circle
- Added Visualisation class

TimPatrickProgramming committed:
Added Description to Architecture and diagrms

- Added a Intro to the Architecture section
- Added Class Responsibility segment to Further describe the Intention of the classes

TimPatrickProgramming committed:
Added Dynamic view sequence diagram

- Finished the Dynamic view Sequence Diagram

TimPatrickProgramming committed:
Update project title in README.md

- Updated the Title for the read me

TimPatrickProgramming committed:
Added descritpion to Dynamic view diagram

- Added a description to further explain the dynamic view sequence diagram

TimPatrickProgramming committed:
Added Additional Documentation

- Added the segment System requirements where different requirements are listed
- For instance the Software Requirements
- The dependencys
- And the Hardware Requirements

TimPatrickProgramming committed:
Merge branch 'Documentation' of https://github.com/TimPatrickProgramming/Software_Enginering_Pattern_Recognition into Documentation

TimPatrickProgramming committed:
Added Table of contents

- Added a table of contents for better overview and navigation
- Adjusted the purpose fo this project segment, to include the reason for developing the project

TimPatrickProgramming committed:
Updated Title of Readme

- The previous changes to the Readme Title seem to have vanished, so they have been done again

Ueli2000 commented:
-  Project Setup, command line arguments are not described and path to main.py is wrong
-  C4 Diagram is missing

TimPatrickProgramming commented:
- Available start commands: 
python python\main.py -> Starts Config version
python python\main.py GUI -> Starts Config version but with GUI
python python\main.py CONSOLE -> Starts Config version but in Consol Mode
python python\main.py CONSOLE CAMERA -> Starts Console Mode with Camera Mode
python python\main.py CONSOLE IMAGE -> Starts Console Mode in Image Mode

TimPatrickProgramming committed:
Added config file descritpion

- Added description about config file and how to use it.
- Added description for start comands and how to use them

TimPatrickProgramming committed:
Updated Dependencies

- Deleted Torch and scipy librarys

TimPatrickProgramming committed:
Started Adding C4 Model

- Added Context Diagramm from C4 Model

TimPatrickProgramming committed:
Added Container Diagram

- Added Container Diagram for C4 Model
- Added elaboration on key architectural decisions

TimPatrickProgramming committed:
Updated Table of contents

- Updated the Table of contents with all the changes made

TimPatrickProgramming commented:
- Added C4 Model
- Updated Various other Sections, see commits for further details
